request.go

```diff
@@ -380,6 +390,87 @@ func (conn *Connection) ExecuteAsync(expr string, args interface{}) *Future {
 	})
 }
 
+// PrepareAsync sends a sql statement to prepare and returns PreparedStatement object.
+//
+// Since 1.6.0
+func (conn *Connection) PrepareAsync(expr string) *PreparedStatement {
+	future := conn.newFuture(PrepareRequest)
+	stmt := newPreparedStatement(future.send(conn, func(enc *msgpack.Encoder) error {
+		enc.EncodeMapLen(1)
+		enc.EncodeUint64(KeySQLText)
+		return enc.EncodeString(expr)
+	}), conn)
+	return stmt
+}
+
+type PreparedStatementID uint64
+
+// PreparedStatement is a type for handling prepared statements
+//
+// Since 1.6.0
+type PreparedStatement struct {
+	StatementID PreparedStatementID
+	MetaData    []ColumnMetaData
+	ParamCount  uint64
+	conn        *Connection
+	fut         *Future
+}
+
+func newPreparedStatement(fut *Future, conn *Connection) *PreparedStatement {
+	stmt := new(PreparedStatement)
+	stmt.fut = fut
+	stmt.conn = conn
+	return stmt
+}
+
+// wait until the prepared statement is ready and fill the statement object
+func (stmt *PreparedStatement) wait() error {
+	resp, err := stmt.fut.Get()
+	stmt.StatementID = PreparedStatementID(resp.StmtID)
+	stmt.MetaData = resp.MetaData
+	stmt.ParamCount = resp.BindCount
```

Comment:
Why we don't call it BindCount too for the consistency?

Reply:
I could. Just repeating the name from [documentation](https://www.tarantool.io/ru/doc/latest/reference/reference_lua/box_sql/prepare/).